---
title: "Literature"
---



The `citep("saha_brief_2013")`

# References

[1] B. E. Blaisdell. "A measure of the similarity of sets of
sequences not requiring sequence alignment." In: _Proceedings of
the National Academy of Sciences of the United States of America_
83.14 (Jul. 1986), pp. 5155-5159. ISSN: 0027-8424. <URL:
http://www.ncbi.nlm.nih.gov/pmc/articles/PMC323909/> (visited on
06/28/2015).

[2] H. Nakashima, M. Ota, K. Nishikawa, et al. "Genes from nine
genomes are separated into their organisms in the dinucleotide
composition space". Eng. In: _DNA research: an international
journal for rapid publication of reports on genes and genomes_ 5.5
(1998), pp. 251-259. ISSN: 1340-2838.

[3] J. T. L. Wang, J. W. Qicheng, D. Shasha, et al. "Application
of Neural Networks to Biological Data Mining: A Case Study in
Protein Sequence Classification". In: _In Proceedings of the Sixth
ACM SIGKDD International Conference on Knowledge Discovery and
Data Mining_. 2000, pp. 305-309.

[4] M. Ganapathiraju, D. Weisser, R. Rosenfeld, et al.
"Comparative N-gram Analysis of Whole-genome Protein Sequences".
In: _Proceedings of the Second International Conference on Human
Language Technology Research_. HLT '02. San Francisco, CA, USA:
Morgan Kaufmann Publishers Inc., 2002, pp. 76-81. <URL:
http://dl.acm.org/citation.cfm?id=1289189.1289259> (visited on
07/31/2015).

[5] F. Tekaia, E. Yeramian and B. Dujon. "Amino acid composition
of genomes, lifestyles of organisms, and evolutionary trends: a
global picture with correspondence analysis". Eng. In: _Gene_
297.1-2 (Sep. 2002), pp. 51-60. ISSN: 0378-1119.

[6] H. H. Otu and K. Sayood. "A new sequence distance measure for
phylogenetic tree construction". En. In: _Bioinformatics_ 19.16
(Nov. 2003), pp. 2122-2130. ISSN: 1367-4803, 1460-2059. DOI:
10.1093/bioinformatics/btg295. <URL:
http://bioinformatics.oxfordjournals.org/content/19/16/2122>
(visited on 06/29/2015).

[7] D. T. Pride, R. J. Meinersmann, T. M. Wassenaar, et al.
"Evolutionary implications of microbial genome tetranucleotide
frequency biases". Eng. In: _Genome Research_ 13.2 (Feb. 2003),
pp. 145-158. ISSN: 1088-9051. DOI: 10.1101/gr.335003.

[8] S. Vinga and J. Almeida. "Alignment-free sequence comparisona
review". En. In: _Bioinformatics_ 19.4 (2003), pp. 513-523. ISSN:
1367-4803, 1460-2059. DOI: 10.1093/bioinformatics/btg005. <URL:
http://bioinformatics.oxfordjournals.org/content/19/4/513>
(visited on 07/31/2015).

[9] K. J. Kalafus, A. R. Jackson and A. Milosavljevic. "Pash:
Efficient Genome-Scale Sequence Anchoring by Positional Hashing".
En. In: _Genome Research_ 14.4 (Apr. 2004), pp. 672-678. ISSN:
1088-9051, 1549-5469. DOI: 10.1101/gr.1963804. <URL:
http://genome.cshlp.org/content/14/4/672> (visited on 07/31/2015).

[10] S. Kurtz, A. Phillippy, A. L. Delcher, et al. "Versatile and
open software for comparing large genomes". En. In: _Genome
Biology_ 5.2 (Jan. 2004), p. R12. ISSN: 1465-6906. DOI:
10.1186/gb-2004-5-2-r12. <URL:
http://genomebiology.com/2004/5/2/R12/abstract> (visited on
06/29/2015).

[11] J. Qi, B. Wang and B. Hao. "Whole proteome prokaryote
phylogeny without sequence alignment: a K-string composition
approach". Eng. In: _Journal of Molecular Evolution_ 58.1 (Jan.
2004), pp. 1-11. ISSN: 0022-2844. DOI: 10.1007/s00239-003-2493-7.

[12] A. Bogan-Marta, I. Pitas and K. Lyroudia. "Statistical Method
of Context Evaluation for Biological Sequence Similarity". En. In:
_Artificial Intelligence in Theory and Practice_. Ed. by M.
Bramer. IFIP International Federation for Information Processing
217. Springer US, 2006, pp. 99-108. ISBN: 978-0-387-34654-0
978-0-387-34747-9. <URL:
http://link.springer.com/chapter/10.1007/978-0-387-34747-9_11>
(visited on 07/31/2015).

[13] A. Tomovic, P. Janicic and V. Keelj. "n-Gram-based
classification and unsupervised hierarchical clustering of genome
sequences". In: _Computer Methods and Programs in Biomedicine_
81.2 (Feb. 2006), pp. 137-153. ISSN: 0169-2607. DOI:
10.1016/j.cmpb.2005.11.007. <URL:
http://www.sciencedirect.com/science/article/pii/S0169260705002361>
(visited on 07/31/2015).

[14] M. Höhl and M. A. Ragan. "Is Multiple-Sequence Alignment
Required for Accurate Inference of Phylogeny?" En. In: _Systematic
Biology_ 56.2 (Apr. 2007), pp. 206-221. ISSN: 1063-5157,
1076-836X. DOI: 10.1080/10635150701294741. <URL:
http://sysbio.oxfordjournals.org/content/56/2/206> (visited on
06/29/2015).

[15] A. Kelil, S. Wang, R. Brzezinski, et al. "CLUSS: Clustering
of protein sequences based on a new similarity measure". En. In:
_BMC Bioinformatics_ 8.1 (Aug. 2007), p. 286. ISSN: 1471-2105.
DOI: 10.1186/1471-2105-8-286. <URL:
http://www.biomedcentral.com/1471-2105/8/286/abstract> (visited on
07/31/2015).

[16] N. S. Mitic, G. M. Pavlovic-Laetic and M. V. Beljanski.
"Could n-gram analysis contribute to genomic island
determination?" In: _Journal of Biomedical Informatics_ 41.6
(2008), pp. 936-943. ISSN: 1532-0464. DOI:
10.1016/j.jbi.2008.03.007. <URL:
http://www.sciencedirect.com/science/article/pii/S1532046408000403>
(visited on 07/31/2015).

[17] B. Chor, D. Horn, N. Goldman, et al. "Genomic DNA k-mer
spectra: models and modalities". En. In: _Genome Biology_ 10.10
(2009), p. R108. ISSN: 1465-6906. DOI: 10.1186/gb-2009-10-10-r108.
<URL: http://genomebiology.com/2009/10/10/R108/abstract> (visited
on 07/31/2015).

[18] S. R. Maetschke, K. S. Kassahn, J. A. Dunn, et al. "A visual
framework for sequence analysis using n-grams and spectral
rearrangement". En. In: _Bioinformatics_ 26.6 (2010), pp. 737-744.
ISSN: 1367-4803, 1460-2059. DOI: 10.1093/bioinformatics/btq042.
<URL: http://bioinformatics.oxfordjournals.org/content/26/6/737>
(visited on 07/31/2015).

[19] G. E. Sims and S. Kim. "Whole-genome phylogeny of Escherichia
coli/Shigella group by feature frequency profiles (FFPs)". En. In:
_Proceedings of the National Academy of Sciences_ 108.20 (2011),
pp. 8329-8334. ISSN: 0027-8424, 1091-6490. DOI:
10.1073/pnas.1105168108. <URL:
http://www.pnas.org/content/108/20/8329> (visited on 06/29/2015).

[20] E. Behnam, M. S. Waterman and A. D. Smith. "A Geometric
Interpretation for Local Alignment-Free Sequence Comparison". In:
_Journal of Computational Biology_ 20.7 (Jul. 2013), pp. 471-485.
ISSN: 1066-5277. DOI: 10.1089/cmb.2012.0280. <URL:
http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3704055/> (visited on
07/31/2015).

[21] S. N. Gardner and B. G. Hall. "When Whole-Genome Alignments
Just Won't Work: kSNP v2 Software for Alignment-Free SNP Discovery
and Phylogenetics of Hundreds of Microbial Genomes". In: _PLoS
ONE_ 8.12 (2013), p. e81760. DOI: 10.1371/journal.pone.0081760.
<URL: http://dx.doi.org/10.1371/journal.pone.0081760> (visited on
07/31/2015).

[22] B. Hofmeister and B. R. King. "ngPhylo: N-Gram Modeled
Proteins with Substitution Matrices for Phylogenetic Analysis".
In: _Proceedings of the International Conference on
Bioinformatics, Computational Biology and Biomedical Informatics_.
BCB'13. New York, NY, USA: ACM, 2013, pp. 676:676-676:677. ISBN:
978-1-4503-2434-2. DOI: 10.1145/2506583.2506679. <URL:
http://doi.acm.org/10.1145/2506583.2506679> (visited on
07/31/2015).

[23] S. Saha and R. Chaki. "A Brief Review of Data Mining
Application Involving Protein Sequence Classification". En. In:
_Advances in Computing and Information Technology_. Ed. by N.
Meghanathan, D. Nagamalai and N. Chaki. Advances in Intelligent
Systems and Computing 177. Springer Berlin Heidelberg, 2013, pp.
469-477. ISBN: 978-3-642-31551-0 978-3-642-31552-7. <URL:
http://link.springer.com/chapter/10.1007/978-3-642-31552-7_48>
(visited on 07/31/2015).

[24] S. M. Srinivasan, S. Vural, B. R. King, et al. "Mining for
class-specific motifs in protein sequence classification". En. In:
_BMC Bioinformatics_ 14.1 (2013), p. 96. ISSN: 1471-2105. DOI:
10.1186/1471-2105-14-96. <URL:
http://www.biomedcentral.com/1471-2105/14/96/abstract> (visited on
07/31/2015).

[25] L. Yang, X. Zhang and H. Zhu. "Alignment free comparison: k
word voting model and its applications". In: _Journal of
Theoretical Biology_ 335 (2013), pp. 276-282. ISSN: 0022-5193.
DOI: 10.1016/j.jtbi.2013.06.037. <URL:
http://www.sciencedirect.com/science/article/pii/S0022519313003202>
(visited on 07/31/2015).

[26] H. Yi and L. Jin. "Co-phylog: an assembly-free phylogenomic
approach for closely related organisms". Eng. In: _Nucleic Acids
Research_ 41.7 (Apr. 2013), p. e75. ISSN: 1362-4962. DOI:
10.1093/nar/gkt003.

[27] O. Bonham-Carter, J. Steele and D. Bastola. "Alignment-free
genetic sequence comparisons: a review of recent approaches by
word analysis". Eng. In: _Briefings in Bioinformatics_ 15.6 (Nov.
2014), pp. 890-905. ISSN: 1477-4054. DOI: 10.1093/bib/bbt052.

[28] B. Haubold. "Alignment-free phylogenetics and population
genetics". Eng. In: _Briefings in Bioinformatics_ 15.3 (2014), pp.
407-418. ISSN: 1477-4054. DOI: 10.1093/bib/bbt083.

[29] S. Horwege, S. Lindner, M. Boden, et al. "Spaced words and
kmacs: fast alignment-free sequence comparison based on inexact
word matches". En. In: _Nucleic Acids Research_ (2014), p. gku398.
ISSN: 0305-1048, 1362-4962. DOI: 10.1093/nar/gku398. <URL:
http://nar.oxfordjournals.org/content/early/2014/05/14/nar.gku398>
(visited on 07/31/2015).

[30] C. Leimeister, M. Boden, S. Horwege, et al. "Fast
alignment-free sequence comparison using spaced-word frequencies".
En. In: _Bioinformatics_ 30.14 (Jul. 2014), pp. 1991-1999. ISSN:
1367-4803, 1460-2059. DOI: 10.1093/bioinformatics/btu177. <URL:
http://bioinformatics.oxfordjournals.org/content/30/14/1991>
(visited on 07/31/2015).

[31] C. Leimeister and B. Morgenstern. "kmacs: the k-mismatch
average common substring approach to alignment-free sequence
comparison". En. In: _Bioinformatics_ 30.14 (Jul. 2014), pp.
2000-2008. ISSN: 1367-4803, 1460-2059. DOI:
10.1093/bioinformatics/btu331. <URL:
http://bioinformatics.oxfordjournals.org/content/30/14/2000>
(visited on 07/31/2015).

[32] B. Liu, J. Xu, Q. Zou, et al. "Using distances between
Top-n-gram and residue pairs for protein remote homology
detection". En. In: _BMC Bioinformatics_ 15.Suppl 2 (Jan. 2014),
p. S3. ISSN: 1471-2105. DOI: 10.1186/1471-2105-15-S2-S3. <URL:
http://www.biomedcentral.com/1471-2105/15/S2/S3/abstract> (visited
on 07/31/2015).

[33] J. B. Pettengill, Y. Luo, S. Davis, et al. "An evaluation of
alternative methods for constructing phylogenies from whole genome
sequence data: a case study with \textitSalmonella". En. In:
_PeerJ_ 2 (2014), p. e620. ISSN: 2167-8359. DOI:
10.7717/peerj.620. <URL: https://peerj.com/articles/620> (visited
on 07/31/2015).

[34] A. Subramanian and R. Schwartz. "Reference-free inference of
tumor phylogenies from single-cell sequencing data". In: _2014
IEEE 4th International Conference on Computational Advances in Bio
and Medical Sciences (ICCABS)_. Jun. 2014, pp. 1-1. DOI:
10.1109/ICCABS.2014.6863944.

[35] J. Wen, R. H. F. Chan, S. Yau, et al. "K-mer natural vector
and its application to the phylogenetic analysis of genetic
sequences". In: _Gene_ 546.1 (Aug. 2014), pp. 25-34. ISSN:
0378-1119. DOI: 10.1016/j.gene.2014.05.043. <URL:
http://www.sciencedirect.com/science/article/pii/S0378111914006064>
(visited on 07/31/2015).

[36] J. Wen, Y. Zhang and S. S. T. Yau. "k-mer Sparse matrix model
for genetic sequence and its applications in sequence comparison".
In: _Journal of Theoretical Biology_ 363 (2014), pp. 145-150.
ISSN: 0022-5193. DOI: 10.1016/j.jtbi.2014.08.028. <URL:
http://www.sciencedirect.com/science/article/pii/S0022519314004895>
(visited on 06/28/2015).

[37] I. Borozan, S. Watt and V. Ferretti. "Integrating
alignment-based and alignment-free sequence similarity measures
for biological sequence classification". En. In: _Bioinformatics_
(Jan. 2015), p. btv006. ISSN: 1367-4803, 1460-2059. DOI:
10.1093/bioinformatics/btv006. <URL:
http://bioinformatics.oxfordjournals.org/content/early/2015/02/04/bioinformatics.btv006>
(visited on 07/31/2015).

[38] S. Xie, P. Li, Y. Jiang, et al. "A discriminative method for
protein remote homology detection based on N-Gram". Eng. In:
_Genetics and molecular research: GMR_ 14.1 (2015), pp. 69-78.
ISSN: 1676-5680. DOI: 10.4238/2015.January.15.9.
